Properly check whether user is locked

NobodysNightmare opened the pull request:
The current implementation opens a small window of opportunity to authenticate through a previously issued access token even after a user account was locked.

If access tokens had a longer lifetime, this could become a large window of opportunity.

# Ticket
* https://community.openproject.org/wp/64883
* Noticed while working on https://github.com/opf/openproject/pull/19206, where we introduce access tokens with a longer lifespan

NobodysNightmare committed:
Properly check whether user is locked

The current implementation opens a small window of
opportunity to authenticate through a previously issued
access token even after a user account was locked.

If access tokens had a longer lifetime, this could become
a large window of opportunity.

ba1ash commented on lib_static/open_project/authentication/strategies/warden/doorkeeper_oauth.rb:
```diff
@@ -43,7 +43,7 @@ def access_token_invalid?(access_token)
 
           def authenticate_user(id)
             user = id && User.find_by(id:)
```
```suggestion
            user = id && User.active.find_by(id:)
```

ba1ash commented on lib_static/open_project/authentication/strategies/warden/doorkeeper_oauth.rb:
```diff
@@ -43,7 +43,7 @@ def access_token_invalid?(access_token)
           def authenticate_user(id)
             user = id && User.find_by(id:)
-            if user
+            if user&.active?
```
```suggestion
            if user
```

ba1ash reviewed (approved):
Good catch! I have one non-critical code suggestion.

NobodysNightmare committed:
Check active state of user during SQL query

This should be a tad faster for the case where the user is not activated, because
the result is already filtered when fetching from the database.

Co-authored-by: Pavel Balashou <ba1ashpash@gmail.com>
